server/controllers/stock/lots.js

```diff
@@ -198,12 +201,80 @@ async function merge(req, res, next) {
       transaction.addQuery(deleteLot, [mergeLotUuid]);
     });
     await transaction.execute();
-    res.sendStatus(200);
+    if (!req.params.internal) {
+      res.sendStatus(200);
+    }
   } catch (error) {
     next(error);
   }
 }
 
+/**
+ * GET /lots/merge/auto
+ *
+ * @description
+ * Finds and merges all lots suitable for automatic merging
+ *  - To qualify, the lots must have the same inventory_uuid,
+ *    label, and expiration date.
+ */
+function autoMerge(req, res, next) {
+  // The first query gets the inventory UUID for each
+  // inventory article with duplicate lots (having the
+  // same label, inventory_uuid, and expiration_date).
+  // (Since these are grouped, only one of the several
+  // lots is given.)
+  const query1 = `
+  SELECT
+    BUID(l.uuid) AS uuid, l.label, l.expiration_date,
+    BUID(i.uuid) AS inventory_uuid, i.text as inventory_text,
+    COUNT(*) as num_duplicates
+  FROM lot l
+  JOIN inventory i ON i.uuid = l.inventory_uuid
+  GROUP BY label, inventory_uuid, expiration_date HAVING num_duplicates > 1
+`;
+  // The second query gets all lots matching the label,
+  // inventory_uuid, and expiration dates
+  const query2 = `
+    SELECT
+      BUID(l.uuid) AS uuid, l.label, l.expiration_date,
+      BUID(i.uuid) AS inventory_uuid, i.text as inventory_text
+    FROM lot l
+    JOIN inventory i ON i.uuid = l.inventory_uuid
+    WHERE l.label=? AND i.uuid=? AND l.expiration_date=DATE(?)
+  `;
+  db.exec(query1, [])
+    .then((rows) => {
+      const numInventories = rows.length;
+      let numLots = rows.reduce((sum, row) => {
+        return sum + row.num_duplicates;
+      }, 0) - numInventories;
+      rows.forEach(async (row) => {
+        db.exec(query2, [row.label, db.bid(row.inventory_uuid), row.expiration_date])
+          .then((lots) => {
+            // Arbitrarily pick first lot and merge the duplicates into it
+            const masterLotUuid = lots[0].uuid;
+            const lotUuids = lots.reduce((list, elt) => {
```

Comment:
> It looks like you are trying to `filter` all lots that are the master lot out of the array. Let's do that! Pretty sure this will give you the same result with clearer syntax:

Actually, I'm trying to do two things at once:  Extract the lot uuid from the duplicate lots but only if they do not match the masterLotUuid.   The suggested code accumulates the lots, not the uuids (I tested it to verify).  So later I would have to do a map on the result.  The reduce allows me to do both at once.